src/main/java/org/cryptomator/cryptofs/health/api/DiagnosticResult.java

```diff
@@ -40,10 +48,28 @@ enum Severity {
 	@Override
 	String toString();
 
+	/**
+	 * A fix for the result.
+	 * <p>
+	 * "Fix" does not imply to restore lost data. It only implies, that the issue leading to this result is resolved.
+	 *
+	 * @param pathToVault path to the root directory of the vault
+	 * @param config the vault config
+	 * @param masterkey the masterkey of the vault
+	 * @param cryptor
+	 * @throws IOException
+	 * @throws UnsupportedOperationException if no fix is implemented for this result
+	 * @deprecated Use {@link #getFix(Path, VaultConfig, Masterkey, Cryptor)} instead
+	 */
+	@Deprecated
 	default void fix(Path pathToVault, VaultConfig config, Masterkey masterkey, Cryptor cryptor) throws IOException {
 		throw new UnsupportedOperationException("Fix for result" + this.getClass() + " not implemented");
 	}
```

Comment:
Maybe only throw UOE off the Optional is empty? Then implementations can drop this method, but API consumers don't notice a change.

Reply:
Implemented in https://github.com/cryptomator/cryptofs/pull/149/commits/3ce839049e80814dd617095b42a4771695a0e243.

I left all fix() method in the result impls for testing, but changed their signature (visibility and parameters).